lib/matplotlib/patches.py

```diff
@@ -1497,6 +1497,94 @@ def get_patch_transform(self):
         return self._patch_transform
 
 
+class Annulus(Patch):
+    """
+    An elliptical annulus.
+    """
+
+    def __str__(self):
```

Comment:
This should probably be `__repr__` then, since it's so close to what a repr format would look like

Reply:
I was following the convention for `__str__` methods of other classes in the module. At the moment we have: 
```
a = Annulus((0.5, 0.5), (0.5, 0.3), 0.1, 45)
repr(a) # '<matplotlib.patches.Annulus object at 0x7fbde40c8c18>'
str(a) # 'Annulus(xy=(0.5, 0.5), r=(0.5, 0.3), width=0.1, angle=45)'
```
which is similar to that of eg. `Ellipse`
```
e = Ellipse((0,0), 1, 2, 30)
repr(e) # '<matplotlib.patches.Ellipse object at 0x7fbdd80e9860>'
str(e) #  'Ellipse(xy=(0, 0), width=1, height=2, angle=30)'
'''

Reply:
That existing behavior strikes me as undesirable, but I suppose fixing it is clearly out of scope for this patch.